pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -166,137 +204,153 @@ func (computeCfg *computeServiceConfig) getManagedVnets() map[string]armnetwork.
 func (computeCfg *computeServiceConfig) getVnetPeers(vnetID string) [][]string {
 	snapshot := computeCfg.resourcesCache.GetSnapshot()
 	if snapshot == nil {
-		azurePluginLogger().V(4).Info("Compute service cache snapshot nil", "type", providerType, "account", computeCfg.account)
+		azurePluginLogger().Info("Cache snapshot nil",
+			"type", providerType, "account", computeCfg.accountNamespacedName)
 		return nil
 	}
+
 	vnetPeersCopy := make([][]string, 0)
-	if peers, ok := snapshot.(*computeResourcesCacheSnapshot).vnetPeers[vnetID]; ok {
+	if peers, ok := snapshot.(*cacheSnapshot).vnetPeers[vnetID]; ok {
 		vnetPeersCopy = deepcopy.Copy(peers).([][]string)
 	}
 	return vnetPeersCopy
 }
 
-func (computeCfg *computeServiceConfig) getVirtualMachines() ([]*virtualMachineTable, error) {
-	filters, hasFilters := computeCfg.getComputeResourceFilters()
-	if !hasFilters {
-		azurePluginLogger().V(1).Info("Fetching vm resources from cloud skipped",
-			"account", computeCfg.account, "resource-filters", "not-configured")
-		return nil, nil
+// getSelectorsConfigured finds the list of CES for which making a cloud api call is required.
+func (computeCfg *computeServiceConfig) getSelectorsConfigured() []*selectorConfig {
+	var cesConfigs []*selectorConfig
+	if len(computeCfg.configChangedSelectors) == 0 {
+		// get all CES for cloud api calls.
+		for _, cesConfig := range computeCfg.selectorConfigMap {
+			cesConfigs = append(cesConfigs, cesConfig)
+		}
+	} else {
+		for _, namespacedName := range computeCfg.configChangedSelectors {
+			if cesConfig, ok := computeCfg.selectorConfigMap[*namespacedName]; ok {
+				cesConfigs = append(cesConfigs, cesConfig)
+			}
+		}
+		computeCfg.configChangedSelectors = nil
 	}
+	return cesConfigs
+}
 
-	if filters == nil {
+// TODO: getVirtualMachines should be able to retrieve VM based on VNET ids as well.
```

Comment:
This is used by UpdateSecurityGroupRules, to fetch Vms for a VNET. 
We need this change right?

Reply:
We have added getAllCachedVirtualMachines() function which gets all virtual machines from snapshot (which imply vms for managed vnets), I do not see any function that would require vm for a specific vnet, could you plz point?